### PR TITLE
python311Packages.xhtml2pdf: fix build

### PR DESCRIPTION
**Pull request by drupol**
- Remove the patch, it was already included in 0.2.16.
- Remove `pythonRelaxDeps`, it's not needed anymore

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by kirillrdy — APPROVED** [review]

File: pkgs/development/python-modules/xhtml2pdf/default.nix [hunk 1]
```diff
@@ -31,21 +30,10 @@ buildPythonPackage rec {
...
-    })
-  ];
-
   nativeBuildInputs = [
     setuptools
   ];
 
-  pythonRelaxDeps = [ "reportlab" ];
-
   propagatedBuildInputs = [
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/xhtml2pdf/default.nix:37]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/xhtml2pdf/default.nix [hunk 2]
```diff
@@ -31,21 +30,10 @@ buildPythonPackage rec {
...
-  patches = [
-    # https://github.com/xhtml2pdf/xhtml2pdf/pull/754
-    (fetchpatch2 {
-      name = "reportlab-compat.patch";
-      url = "https://github.com/xhtml2pdf/xhtml2pdf/commit/1252510bd23b833b45b4d252aeac62c1eb51eeef.patch";
-      hash = "sha256-9Fkn086uh2biabmiChbBna8Q4lJV/604yX1ng9j5TGs=";
-    })
-  ];
-
   nativeBuildInputs = [
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/xhtml2pdf/default.nix:33]:
```suggestion
  build-system = [
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/xhtml2pdf/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/python-modules/xhtml2pdf/default.nix]: fixed.

File: pkgs/development/python-modules/xhtml2pdf/default.nix (hunk 2 above)
Comment (drupol) [reply pkgs/development/python-modules/xhtml2pdf/default.nix:33]: Fixed.

Comment (GaetanLepage) [reply pkgs/development/python-modules/xhtml2pdf/default.nix:33]: not sure

Comment (drupol) [reply pkgs/development/python-modules/xhtml2pdf/default.nix]: Sorry it's late !

**Review by GaetanLepage — APPROVED** [review]
LGTM :)

**Pull request merged**